### PR TITLE
fix(griffin): Implement negative period.

### PR DESCRIPTION
**Pull request by eugenels**
It's symmetric to positive one.
e.g. '1970-03-20T01;2h' <=> '1970-03-20T03;-2h'

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/model/IntervalUtils.java|9|9|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/model/IntervalUtils.java|9|9|100.00%|

File: core/src/test/java/io/questdb/griffin/WhereClauseParserTest.java [hunk 1]
```diff
@@ -382,6 +382,46 @@ public void testComplexInterval5() throws Exception {
...
+    @Test
+    public void testComplexIntervalNegMs3() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55.000Z;-4h'", "[{lo=2015-02-23T06:00:55.000000Z, hi=2015-02-23T10:00:55.000000Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNegMs4() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55.000Z;-7d'", "[{lo=2015-02-16T10:00:55.000000Z, hi=2015-02-23T10:00:55.000000Z}]");
+    }
+
```
Comment (ideoma) [inline core/src/test/java/io/questdb/griffin/WhereClauseParserTest.java:424]: Would be good to add tests with 4 token intervals, like 

`'2015-02-23T10:00:55.000Z;-7d;15d;5'`

and

`'2015-02-23T10:00:55.000Z;-7d;-15d;5'`

Both 15d and -15d above doesn't make too much sense to me tbh, perhaps 4 token period with negative intervals should not be valid.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/model/IntervalUtils.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/model/IntervalUtils.java|11|11|100.00%|

**Comment by eugenels** [comment]
replaced by Throw invalid date exception on negative time interval #1277

**Pull request closed without merging**